x/oracle/keeper/historic_price.go

```diff
@@ -187,6 +296,30 @@ func (k Keeper) IterateHistoricPrices(
 	}
 }
 
+// IterateMedians iterates over median prices of a given
+// denom in the store and tracks their block stamp.
+// Iterator stops when exhausting the source, or when the handler returns `true`.
+func (k Keeper) IterateMedians(
+	ctx sdk.Context,
+	denom string,
+	handler func(sdk.Dec, uint64) bool,
+) {
+	store := ctx.KVStore(k.storeKey)
+
+	// make sure we have one zero byte to correctly separate denoms
+	prefix := util.ConcatBytes(1, types.KeyPrefixMedian, []byte(denom))
```

Comment:
we need to have separator. Otherwise the code is not valid. Consider 2 denoms: `eth`, ethpow`. When searching for `eth` one, `ethpow` will be returned as well.

Reply:
I believe this already taken care of by the util methods `ConcatBytes()` and `UintWithNullPrefix()`. I updated `historic_price_test.go` to test for this and it looks like this edge case is covered for.

Reply:
yes, now `ConcatBytes` solves this.